### PR TITLE
[builder] Upload all sc-elements to concerted kb result structure

### PR DESCRIPTION
**Pull request by smousy**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/docs/dev/pr.md/)
* [x] Update changelog
* [x] Update documentation

### CODE REVIEW

File: sc-tools/sc-builder/src/scs_translator.cpp [hunk 1]
```diff
@@ -114,9 +114,19 @@ bool SCsTranslator::TranslateImpl(Params const & params)
 
   SCsHelper scs(m_ctx, std::make_shared<impl::FileProvider>(params.m_fileName));
   
-  if (!scs.GenerateBySCsText(data))
+  if(params.m_resultStructureUpload)
   {
-    SC_THROW_EXCEPTION(utils::ExceptionParseError, scs.GetLastError());
+      if (!scs.GenerateBySCsText(data, params.m_outputStructure))
```
Comment (KovalM) [inline sc-tools/sc-builder/src/scs_translator.cpp:119]: I think we can leave only this "if". We can call scs.GenerateBySCsText(data, params.m_outputStructure), this method will ignore second param if it's not valid

**Review by KovalM — CHANGES_REQUESTED** [review]

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: sc-tools/sc-config-utils/sc_builder_config.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  BuilderParams GetParams()
+  {
+    m_builderParams.m_resultStructureUpload =
+        m_params.count("result_structure_upload")
+            ? (m_params.at("result_structure_upload") == "true" ? SC_TRUE : SC_FALSE)
+            : DEFAULT_RESULT_STRUCTURE_UPLOAD;
+    m_builderParams.m_resultStructureSystemIdtf = m_params.count("result_structure_system_idtf")
+                                                      ? m_params.at("result_structure_system_idtf").c_str()
+                                                      : DEFAULT_RESULT_STRUCTURE_MAIN_IDT;
```
Comment (NikitaZotov) [inline sc-tools/sc-config-utils/sc_builder_config.hpp:49]:
```suggestion
                                                      : DEFAULT_RESULT_STRUCTURE_MAIN_IDTF;
```

File: sc-tools/sc-config-utils/sc_builder_config.hpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    return m_builderParams;
+  }
+
+  ~ScBuilderConfig()
+  {
+  }
```
Comment (NikitaZotov) [inline sc-tools/sc-config-utils/sc_builder_config.hpp:56]:
```suggestion
  ~ScBuilderConfig() = default;
```

File: sc-tools/sc-config-utils/sc_memory_config.hpp [hunk 4]
```diff
@@ -104,7 +104,7 @@ class ScMemoryConfig
         SC_MACHINE_VERSION_MAJOR, SC_MACHINE_VERSION_MINOR, SC_MACHINE_VERSION_PATCH, SC_MACHINE_VERSION_SUFFIX};
 
     m_memoryParams.version = (sc_char const *)sc_version_string_new(&version);
-    m_memoryParams.clear = m_params.count("clear") ? SC_TRUE : SC_FALSE;
+    m_memoryParams.clear = m_params.count("clear") ? (m_params.at("clear") == "true" ? SC_TRUE : SC_FALSE) : SC_FALSE;
```
Comment (NikitaZotov) [inline sc-tools/sc-config-utils/sc_memory_config.hpp:107]: Add info about this fix into changelog

File: sc-tools/sc-builder/tests/units/test_base.cpp [hunk 5]
```diff
@@ -22,10 +23,78 @@ TEST(ScBuilder, RunStop)
...
+  {
+    ScConfigGroup group = configFile[memoryGroupName];
+    for (auto const & key : *group)
+    {
+      std::string const & value = group[key];
+      std::stringstream stream;
+
+      scParams.insert({key, value});
+    }
+    sc_memory_params scMemoryParams = memoryConfig.GetParams();
```
Comment (NikitaZotov) [inline sc-tools/sc-builder/tests/units/test_base.cpp:45]:
```suggestion
    sc_memory_params const scMemoryParams = memoryConfig.GetParams();
```

File: docs/changelog.md [hunk 6]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Append generated elements by `SCs-helper` into requested output structure
 - Oriented sets representation (<a, ..., b>) in SCs
+- Add output structure builder feature to config
```
Comment (NikitaZotov) [inline docs/changelog.md:14]: Move it into the begin of this section, please. The first messages of each section show what was changed last, and the last messages show what was done the firstly

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

**Review by KovalM — APPROVED** [review]

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**